src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -2122,8 +2122,8 @@ bool Lowering::ContainCheckCompareChain(GenTree* child, GenTree* parent, GenTree
         return true;
     }
     // Can the child be contained.
-    else if (IsSafeToContainMem(parent, child) && varTypeIsIntegral(child->gtGetOp1()) &&
-             varTypeIsIntegral(child->gtGetOp2()))
+    else if (IsSafeToContainMem(parent, child) && (child->OperIs(GT_AND) || child->OperIsCmpCompare()) &&
+             varTypeIsIntegral(child->gtGetOp1()) && varTypeIsIntegral(child->gtGetOp2()))
```

Comment:
Can this just be done in the `OperIsCmpCompare()` case below?

Reply:
(like you have it in `IsValidCompareChain`)

Reply:
I don't think we would expect `GT_AND` with float operands -- seems like invalid IR.

Reply:
Yea that's fair.